lms/djangoapps/courseware/models.py

```diff
@@ -248,3 +254,95 @@ class Meta(object):   # pylint: disable=missing-docstring
 
     field = models.CharField(max_length=255)
     value = models.TextField(default='null')
+
+
+SCORE_CHANGED = Signal(
+    providing_args=[
+        'points_possible',  # Maximum score available for the exercise
+        'points_earned',   # Score obtained by the user
+        'user',  # Integer User ID
+        'course',  # Unicode string representing the course
+        'usage'  # Unicode string indicating the courseware instance
+    ]
+)
+
+
+@receiver(score_set)
+def submissions_score_set_handler(sender, **kwargs):  # pylint: disable=unused-argument
+    """
+    Consume the score_set signal defined in the Submissions API, and convert it
+    to a SCORE_CHANGED signal defined in this module. Converts the unicode keys
+    for user, course and item into the standard representation for the
+    SCORE_CHANGED signal.
+
+    This method expects that the kwargs dictionary will contain the following
+    entries (See the definition of score_set):
+      - 'points_possible': integer,
+      - 'points_earned': integer,
+      - 'anonymous_user_id': unicode,
+      - 'course_id': unicode,
+      - 'item_id': unicode
+    """
+    points_possible = kwargs.get('points_possible', None)
+    points_earned = kwargs.get('points_earned', None)
+    course = kwargs.get('course_id', None)
+    usage = kwargs.get('item_id', None)
```

Comment:
Just as a note, `item_id` was intentionally named differently because submissions knows nothing about XBlocks. It was originally envisioned as a way to put things into submissions and assign them scores without them existing in the course tree -- e.g. "class participation". We might have to re-evaluate that and make it aware of opaque-keys.

This is not a request for action, I just wanted to give context.